cns/service/main.go

```diff
@@ -535,6 +535,14 @@ func main() {
 		}
 	}
 
+	go func() {
+		// Periodically poll NC version from NMAgent
+		for {
```

Comment:
If this thread crashes, how do you restart?

Reply:
Restart is taken care by the next line time.NewTicker. Ticker will let us do something repeatedly at regular intervals.